stl/inc/chrono

```diff
@@ -619,8 +618,61 @@ namespace chrono {
             // but the initial value of _Ctr could be large.
             const long long _Whole = (_Ctr / _Freq) * period::den;
             const long long _Part  = (_Ctr % _Freq) * period::den / _Freq;
-            return time_point(duration(_Whole + _Part));
+            return _Whole + _Part;
+        }
+
+        _NODISCARD static time_point now() noexcept { // get current time
+#if (defined(_M_IX86) || defined(_M_X64) || defined(_M_ARM) || defined(_M_ARM64)) && !defined(_M_CEE_PURE)
+            // Implement atomics avoiding <atomic> header dependency
+            static struct {
+                volatile long long _Freq        = LLONG_MAX;
+                volatile long long _Ctr_base    = LLONG_MAX;
+                volatile long long _Result_base = LLONG_MAX;
+            } _Cached;
+            const long long _Freq_from_cache = _Atomic_load_ll_relaxed(&_Cached._Freq);
+            const long long _Ctr_base        = _Atomic_load_ll_relaxed(&_Cached._Ctr_base);
+            const long long _Result_base     = _Atomic_load_ll_relaxed(&_Cached._Result_base);
+            if (_Freq_from_cache != LLONG_MAX && _Ctr_base != LLONG_MAX && _Result_base != LLONG_MAX) {
+                // Fast path
+                const long long _Ctr = _Query_perf_counter();
+                return time_point(duration((_Ctr - _Ctr_base) * period::den / _Freq_from_cache + _Result_base));

```

Comment:
Would this be clearer?

```suggestion
                return time_point(duration(_Result_base + (_Ctr - _Ctr_base) * period::den / _Freq_from_cache));
```

In addition to "starting with" `_Result_base`, it avoids the somewhat-visually-confusing-to-humans pattern of `A / B + C`.